merkledag.go

```diff
@@ -337,27 +394,7 @@ func (p *ProgressTracker) Value() int {
 	return p.Total
 }
 
-// FetchGraphConcurrency is total number of concurrent fetches that
-// 'fetchNodes' will start at a time
-var FetchGraphConcurrency = 32
-
-// WalkParallel is equivalent to Walk *except* that it explores multiple paths
-// in parallel.
-//
-// NOTE: It *does not* make multiple concurrent calls to the passed `visit` function.
-func WalkParallel(ctx context.Context, getLinks GetLinks, c cid.Cid, visit func(cid.Cid) bool) error {
-	visitDepth := func(c cid.Cid, depth int) bool {
-		return visit(c)
-	}
-
-	return WalkParallelDepth(ctx, getLinks, c, 0, visitDepth)
-}
-
-// WalkParallelDepth is equivalent to WalkDepth *except* that it fetches
-// children in parallel.
-//
-// NOTE: It *does not* make multiple concurrent calls to the passed `visit` function.
-func WalkParallelDepth(ctx context.Context, getLinks GetLinks, c cid.Cid, startDepth int, visit func(cid.Cid, int) bool) error {
+func parallelWalkDepth(ctx context.Context, getLinks GetLinks, root cid.Cid, visit func(cid.Cid, int) bool, options *WalkOptions) error {
```

Comment:
note: I kept this function at the same place as before for an easier review, but it would make sense to move it just after `sequentialWalkDepth`.